bundles/org.eclipse.passage.lic.bc/src/org/eclipse/passage/lic/internal/bc/BcStreamCodec.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.internal.bc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.eclipse.passage.lic.internal.api.LicensedProduct;
+import org.eclipse.passage.lic.internal.api.io.DigestExpectation;
+import org.eclipse.passage.lic.internal.api.io.EncryptionAlgorithm;
+import org.eclipse.passage.lic.internal.api.io.EncryptionKeySize;
+import org.eclipse.passage.lic.internal.api.io.StreamCodec;
+
+@SuppressWarnings("restriction")
+public final class BcStreamCodec implements StreamCodec {
+
+	private final LicensedProduct product;
+	private final EncryptionAlgorithm algorithm;
+	private final EncryptionKeySize keySize;
+
+	public BcStreamCodec(LicensedProduct product, EncryptionAlgorithm algorithm, EncryptionKeySize keySize) {
+		this.product = product;
+		this.algorithm = algorithm;
+		this.keySize = keySize;
+	}
+
+	@Override
+	public LicensedProduct id() {
+		return product;
+	}
+
+	@Override
+	public EncryptionAlgorithm algorithm() {
+		return algorithm;
+	}
+
+	@Override
+	public EncryptionKeySize keySize() {
+		return keySize;
+	}
+
+	@Override
+	public void createKeyPair(String publicKeyPath, String privateKeyPath, String username, String password)
+			throws IOException {
+		// TODO Auto-generated method stub
```

Comment:
It does nothing here and below. Is this intentional?

Reply:
It has just a stub - i've just started the odec redesign. The following request is going contain reworked implementation.